### DISCUSSION TITLE
Text field with new line

### DISCUSSION THREAD

**Discussion by royaljeffr** [Budibase/budibase#4759] [Help]
I would like to have a text field but have the option for a new line (carriage return) so when it gets viewed somewhere else it will be multiline.

**Comment by gerhardmpl** [comment]
You could use the `Long Form Text` data type instead of the `Text` data type when creating the column in the Data Section. This will also give you the rich text editor.

**Reply by royaljeffr** [reply]
I am pulling the data from a Postgres database.  Do I need to change the data type at the DB level and then re-add the dataset to Budibase?

**Reply by gerhardmpl** [reply]
Unfortunately I don't know because I only use `bbdb`. Maybe one of the Maintainer can help?

**Reply by mjashanks** [reply]
@royaljeffr you shouldn't need to change your column at the DB level. Just use a `Long form field` control where you need it.

**Reply by royaljeffr** [reply]
On the Data tab Budibase shows the column type as Text and I can't change it there.  And within Design if I go to Form and look at my list of controls I can place on the form, Long form field is not there.

**Reply by mjashanks** [reply]
Strange - are you running self-hosted? If so have you updated recently?

My list of form controls:

![image](https://user-images.githubusercontent.com/3524181/156606643-2cf5cc7a-8128-4fed-8ba4-8047f8a49e40.png)

What does yours look like?

You might have to click this little button, if it's blue:

![image](https://user-images.githubusercontent.com/3524181/156606814-36be0987-7498-4140-bcd9-2c436ea77d7e.png)

This will update the current app to the latest version.

**Reply by royaljeffr** [reply]
That was it.  I did a docker-compose pull, docker-compose restart, and then updated the app with the Blue button and now it's there.  Thank you!

**Reply by royaljeffr** [reply]
But now I get this error
![image](https://user-images.githubusercontent.com/49763873/156608604-55c5f6a3-909a-45e4-be39-8718179e3a21.png)

**Reply by mjashanks** [reply]
Hmm, try editing your column and selecting "Enable rich text support"

![image](https://user-images.githubusercontent.com/3524181/156609709-2dbc1562-cba5-453b-b071-14e6493734de.png)

**Reply by royaljeffr** [reply]
I don't have the option to change the column in the Data area to Long Form Text 
![image](https://user-images.githubusercontent.com/49763873/156610095-9cfd2aed-e40e-4329-bb73-8e306c93e6d7.png)

**Reply by royaljeffr** [reply]
I am on version 1.0.73

**Reply by mjashanks** [reply]
Hmm, thats a bit annoying. Ok, for now your options are

Change your DB column to `text` and refetch tables

OR

In your screen, 
- add a rich text component
- manually set the field name to a new name... say `description_long`
- in your `Save` button, manually map the `description` field to the `Form.Fields.description_long` field

as below:

![image](https://user-images.githubusercontent.com/3524181/156612053-221404d9-8311-4f75-af3c-39c736fb5b3f.png)

**Reply by royaljeffr** [reply]
I did option A and things work great.  Thank you for the help!
